src/dev/flang/ast/InlineArray.java

```diff
@@ -181,7 +181,7 @@ private AbstractType elementType(AbstractType t)
     if (PRECONDITIONS) require
       (t != null);
 
-    if (t.featureOfType() == Types.resolved.f_array &&
+    if (Types.resolved.f_array.inheritsFrom(t.featureOfType()) &&
         t.generics().size() == 1)
```

Comment:
Since `Sequence` is the only parent of `array` with one argument, this works, but it is a bit brittle since changes to the parents of `array` may break this code (say, we make `array` inherit from `has_size i32`).

For now, this code is goo, but please add a `NYI` comment here and refer to #1817 such that this can be improved once a more powerful mechanism is in place.